Lib/subprocess.py

```diff
@@ -1480,7 +1480,20 @@ def _execute_child(self, args, executable, preexec_fn, close_fds,
             if shell:
                 startupinfo.dwFlags |= _winapi.STARTF_USESHOWWINDOW
                 startupinfo.wShowWindow = _winapi.SW_HIDE
-                comspec = os.environ.get("COMSPEC", "cmd.exe")
+                # gh-101283: without a fully-qualified path, before Windows
+                # checks the system directories, it first looks in the
+                # application directory, and also the current directory if
+                # NeedCurrentDirectoryForExePathW(ExeName) is true, so try
+                # to avoid executing unqualified "cmd.exe".
+                comspec = os.environ.get('ComSpec')
+                if not comspec:
+                    system_root = os.environ.get('SystemRoot')
+                    comspec = os.path.join(system_root, 'System32', 'cmd.exe')
+                    if not os.path.isfile(comspec):
```

Comment:
```suggestion
                    if not os.path.isabs(comspec) or not os.path.isfile(comspec):
```

Avoid testing a relative path at all and just raise the error. We'll still need the isabs check on 1494 because it's come through a different code path.